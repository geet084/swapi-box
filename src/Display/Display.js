import React from 'react'
import Card from '../Card/Card'
import './Display.scss'
import PropTypes from 'prop-types'

const Display = ({ choice, selection }) => {
  
  if (choice !== '' && choice !== 'faves') {
    let cards = selection.map((selected, i) => {
      if (selected.residents && selected.residents.length < 1) {
        selected.residents = ['none']
      }
      return <Card key={i} choice={choice} {...selected} />
    })

    return (
      <div className="display">
        {cards}
      </div>
    )
  } else if (choice === 'faves') { 
    return (
      <div>
      </div>
    )
  } else {
    return (
    <div className="select-category">
      <span className="select">SELECT A CATEGORY</span>
      <span className="category"></span>
    </div>
    )
  }

}

Display.protoTypes = {
  choice: PropTypes.string,
  selection: PropTypes.string
}
export default Display;