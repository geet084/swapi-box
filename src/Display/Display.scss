.display {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-evenly;
}

.select-category {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 15%;
  font-size: 3rem;
  color: white;
}
