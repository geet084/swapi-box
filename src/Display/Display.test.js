import React from 'react'
import { shallow } from 'enzyme'
import Display from './Display'

describe('Display', () => {
  let wrapper;

  it('should properly render default snapshot', () => {
    const mockChoice = ''
    const mockThing = { choice: mockChoice, vehicles: [] }
    wrapper = shallow(
      <Display
        choice={mockChoice}
        {...mockThing} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should properly render a chosen category', () => {
    const mockChoice = 'vehicles'
    const mockThing = [{}] 
    wrapper = shallow(
      <Display
        choice={mockChoice}
        selection={mockThing} />
    )

    expect(wrapper).toMatchSnapshot();
  })
})