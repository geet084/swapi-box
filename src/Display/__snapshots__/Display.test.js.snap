// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should properly render a chosen category 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display
    choice="vehicles"
    selection={
      Array [
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          choice="vehicles"
        />,
      ],
      "className": "display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "choice": "vehicles",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            choice="vehicles"
          />,
        ],
        "className": "display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "choice": "vehicles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Display should properly render default snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display
    choice=""
    vehicles={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="select"
        >
          SELECT A CATEGORY
        </span>,
        <span
          className="category"
        />,
      ],
      "className": "select-category",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SELECT A CATEGORY",
          "className": "select",
        },
        "ref": null,
        "rendered": "SELECT A CATEGORY",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "category",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="select"
          >
            SELECT A CATEGORY
          </span>,
          <span
            className="category"
          />,
        ],
        "className": "select-category",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SELECT A CATEGORY",
            "className": "select",
          },
          "ref": null,
          "rendered": "SELECT A CATEGORY",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "category",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
